core/arch/arm/plat-imx/psci.c

```diff
@@ -67,3 +71,68 @@ int psci_cpu_on(uint32_t core_idx, uint32_t entry,
 
 	return PSCI_RET_SUCCESS;
 }
+
+int psci_cpu_off(void)
+{
+	uint32_t core_id;
+	vaddr_t va = core_mmu_get_va(SRC_BASE, MEM_AREA_IO_SEC);
+
+	core_id = get_core_pos();
+
+	write32(~0, va + SRC_GPR1 + core_id * 8 + 4);
+
+	DMSG("%d\n", core_id);
+	psci_armv7_cpu_off();
+
+	imx_set_src_gpr(core_id, ~0);
+
+	asm volatile (
+		"mrs	r0, cpsr\r\n"
+		"orr	r0, r0, #(1 << 6)\r\n"
+		"orr	r0, r0, #(1 << 7)\r\n"
+		"msr	cpsr, r0\r\n"
+		"loop:\r\n"
+		"wfi\r\n"
+		" b loop\r\n");
+
+	return PSCI_RET_SUCCESS;
+}
+
+int psci_affinity_info(uint32_t affinity,
+		       uint32_t lowest_affnity_level __unused)
+{
+	vaddr_t va = core_mmu_get_va(SRC_BASE, MEM_AREA_IO_SEC);
+	vaddr_t gpr5 = core_mmu_get_va(IOMUXC_BASE, MEM_AREA_IO_SEC) +
+		IOMUXC_GPR5_OFFSET;
+	uint32_t cpu, val;
+	uint32_t wfi;
+
+	cpu = affinity;
+
+	wfi = !!(read32(gpr5) & ARM_WFI_STAT_MASK(cpu));
+
+	if ((imx_get_src_gpr(cpu) == 0) || !wfi)
```

Comment:
This is not to check false or true, I just check whether its value is 0 or `non-zero`, I prefer use `==` here.